src/main/java/org/stellar/sdk/AccountConverter.java

```diff
@@ -0,0 +1,41 @@
+package org.stellar.sdk;
+
+import org.stellar.sdk.xdr.CryptoKeyType;
+import org.stellar.sdk.xdr.MuxedAccount;
+
+public class AccountConverter {
+  private final boolean enableMuxed;
+  private AccountConverter(boolean enabled) {
+    this.enableMuxed = enabled;
+  }
+
+  public static AccountConverter enableMuxed() {
+    return new AccountConverter(true);
+  }
+
+  public static AccountConverter disableMuxed() {
+    return new AccountConverter(false);
+  }
+
+
+  public MuxedAccount encode(String account) {
+    MuxedAccount muxed = StrKey.encodeToXDRMuxedAccount(account);
+
+    if (this.enableMuxed || muxed.getDiscriminant().equals(CryptoKeyType.KEY_TYPE_ED25519)) {
+      return muxed;
+    }
+
+    MuxedAccount.MuxedAccountMed25519 med25519 = muxed.getMed25519();
+    muxed.setDiscriminant(CryptoKeyType.KEY_TYPE_ED25519);
+    muxed.setEd25519(med25519.getEd25519());
+    return muxed;
+  }
+
+  public String decode(MuxedAccount account) {
+    if (this.enableMuxed || account.getDiscriminant().equals(CryptoKeyType.KEY_TYPE_ED25519)) {
+      return StrKey.encodeStellarMuxedAccount(account);
+    }
+
+    return StrKey.encodeStellarAccountId(StrKey.muxedAccountToAccountId(account));
```

Comment:
Just noting here that this is the same behavior as in the [JS SDK](https://github.com/stellar/js-stellar-base/blob/1145458b490a1e3aa1275cd257b35010ea487381/src/util/decode_encode_muxed_account.js#L60-L71): force a fully-muxed account to a G-address when `enableMuxed = false` :heavy_check_mark: